src/Protocol/Diaspora.php

```diff
@@ -3850,20 +3850,26 @@ private static function constructComment(array $item, array $owner)
 			return $result;
 		}
 
-		$toplevel_item = Item::selectFirst(['guid', 'author-link'], ['id' => $item["parent"], 'parent' => $item["parent"]]);
+		$toplevel_item = Item::selectFirst(['guid', 'author-id', 'author-link'], ['id' => $item["parent"], 'parent' => $item["parent"]]);
 		if (!DBA::isResult($toplevel_item)) {
 			Logger::error('Missing parent conversation item', ['parent' => $item["parent"]]);
 			return false;
 		}
 
 		$thread_parent_item = $toplevel_item;
 		if ($item['thr-parent'] != $item['parent-uri']) {
-			$thread_parent_item = Item::selectFirst(['guid', 'author-link'], ['uri' => $item['thr-parent'], 'uid' => $item['uid']]);
+			$thread_parent_item = Item::selectFirst(['guid', 'author-id', 'author-link'], ['uri' => $item['thr-parent'], 'uid' => $item['uid']]);
 		}
 
 		$body = $item["body"];
 
-		if ((empty($item['uid']) || !Feature::isEnabled($item['uid'], 'explicit_mentions'))
+		// The replied to autor mention is prepended for clarity if:
+		// - Item replied isn't yours
+		// - Item is public or explicit mentions are disabled
+		// - Implicit mentions are enabled
+		if (
+			$owner['id'] != $thread_parent_item['author-id']
```

Comment:
The author-id is the value for the public contact. This differs from the id in the owner array. The item should have the author-id as well, so you could compare these two.

Reply:
Got it, thanks.